### ISSUE TITLE
We need documentation how to use kotlin.parallel.tasks.in.project=true

### ISSUE THREAD

**Issue by pe4enko** [gradle/gradle#17231]
Long ago, when Kotlin version 1.3.20 was released (https://blog.jetbrains.com/kotlin/2019/01/kotlin-1-3-20-released/), the ability to build in parallel using Gradle Workers was added. Simply adding the **kotlin.parallel.tasks.in.project = true** setting does not give any gain in build speed. As far as I understand, this parameter can be useful only if I have several folders with classes independent of each other within the same project. I saw the use of this setting when assembling the **gradle** itself, but did not see anywhere that separate source sets were created for each folder.
Could you provide examples of how to correctly describe the build process in **build.gradle.kts** so that mentioned option is really used and gives an increase in build speed when there are several processor cores.

**Comment by mlopatkin** [comment]
Thank you for your interest in Gradle!

Your issue appears to be a question about how to use Gradle. Your question might be better answered on our forums or StackOverflow. We have several resources on [help.gradle.org](https://help.gradle.org) that you should check out first. In addition, this information might be available in the [Kotlin plugin's documentation by Jetbrains](https://kotlinlang.org/docs/gradle.html).

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

If you think Gradle is missing a feature, please file a new issue describing your use case for the feature.
